### PR TITLE
Minor fixes to rancher-monitoring v0.1.3

### PR DESCRIPTION
**Pull request by aiyengar2**
Introduces quick fixes for Monitoring V1.

Blocked on merge for PRs affecting previous version:
- [x] Update rancher-monitoring images (https://github.com/rancher/system-charts/pull/334)

Related Issues: https://github.com/rancher/rancher/issues/26902, https://github.com/rancher/rancher/issues/27289

### CODE REVIEW

File: charts/rancher-monitoring/v0.1.4/Chart.yaml
Comment (aiyengar2) [inline charts/rancher-monitoring/v0.1.4/Chart.yaml]: Does the `rancher_min_version` also need to be modified? It's currently at `2.4.6-rc1`.

**Review by aiyengar2 — COMMENTED** [review]
Not sure if these changes require a brand new v0.1.4 chart (as it is currently structured) or if any other changes are required for system-charts PRs. Please let me know if anything looks off since this is my first PR to this repo!

If a brand new chart is not required, I could split this PR into two separate ones that deal with the above issues separately.

File: charts/rancher-monitoring/v0.1.4/Chart.yaml
Comment (deniseschannon) [reply charts/rancher-monitoring/v0.1.4/Chart.yaml]: Can we update it to v2.5.0-alpha1?

Comment (aiyengar2) [reply charts/rancher-monitoring/v0.1.4/Chart.yaml]: done

**Comment by deniseschannon** [comment]
@PennyScissors You should see how this might impact your image changes that you're looking to add.

**Comment by aiyengar2** [comment]
@PennyScissors I'm going to block this merge on https://github.com/rancher/system-charts/pull/334 since it seems we'll need to repeat the base copy step after you merge in changes to 0.1.3 to avoid regressing in 0.1.4.

**Comment by aiyengar2** [comment]
> @PennyScissors I'm going to block this merge on #334 since it seems we'll need to repeat the base copy step after you merge in changes to 0.1.3 to avoid regressing in 0.1.4.

Merge / review is no longer blocked.

I also realized that this PR was done against the `rancher/dev-v2.5` branch instead of `rancher/dev-v2.4` so I modified the base that needs to be updated for this PR as well.

To make it easier to review, here are the actual changes that happen between `v0.1.3` and `v0.1.4` presented diff-style:

```shell
arvindiyengar: ~/Rancher/system-charts/src/github.com/rancher/system-charts 
$ diff -uNr charts/rancher-monitoring/v0.1.3 charts/rancher-monitoring/v0.1.4
```
```diff
diff -uNr charts/rancher-monitoring/v0.1.3/Chart.yaml charts/rancher-monitoring/v0.1.4/Chart.yaml
--- charts/rancher-monitoring/v0.1.3/Chart.yaml	2020-08-06 14:20:41.000000000 -0700
+++ charts/rancher-monitoring/v0.1.4/Chart.yaml	2020-09-17 17:58:14.000000000 -0700
@@ -7,8 +7,8 @@
 name: rancher-monitoring
 sources:
 - https://github.com/coreos/prometheus-operator
-version: 0.1.3
-appVersion: 0.1.3
+version: 0.1.4
+appVersion: 0.1.4
 home: https://github.com/coreos/prometheus-operator
 keywords:
 - operator
diff -uNr charts/rancher-monitoring/v0.1.3/charts/grafana/dashboards/c_kubernetes-components-dashboard.json charts/rancher-monitoring/v0.1.4/charts/grafana/dashboards/c_kubernetes-components-dashboard.json
--- charts/rancher-monitoring/v0.1.3/charts/grafana/dashboards/c_kubernetes-components-dashboard.json	2020-08-06 14:20:41.000000000 -0700
+++ charts/rancher-monitoring/v0.1.4/charts/grafana/dashboards/c_kubernetes-components-dashboard.json	2020-09-17 17:58:14.000000000 -0700
@@ -1 +1 @@
-{"__requires":[{"type":"grafana","id":"grafana","name":"Grafana","version":"5.3.0"},{"type":"panel","id":"graph","name":"Graph","version":"5.0.0"},{"type":"datasource","id":"prometheus","name":"Prometheus","version":"5.0.0"}],"annotations":{"list":[{"builtIn":1,"datasource":"-- Grafana --","enable":true,"hide":true,"iconColor":"rgba(0, 211, 255, 1)","name":"Annotations & Alerts","type":"dashboard"}]},"description":"-","editable":true,"gnetId":5508,"graphTooltip":0,"id":null,"links":[],"panels":[{"aliasColors":{},"bars":false,"dashLength":10,"dashes":false,"datasource":"RANCHER_MONITORING","editable":true,"error":false,"fill":1,"grid":{"threshold1Color":"rgba(216, 200, 27, 0.27)","threshold2Color":"rgba(234, 112, 112, 0.22)"},"gridPos":{"h":7,"w":12,"x":0,"y":0},"id":6,"isNew":false,"legend":{"alignAsTable":false,"avg":false,"current":false,"hideEmpty":false,"hideZero":false,"max":false,"min":false,"rightSide":false,"show":true,"total":false,"values":false},"lines":true,"linewidth":1,"links":[],"nullPointMode":"null","percentage":false,"pointradius":5,"points":false,"renderer":"flot","seriesOverrides":[],"spaceLength":10,"stack":false,"steppedLine":false,"targets":[{"expr":"sum by(instance) (rate(apiserver_request_count{code!~\"2..\"}[5m]))","format":"time_series","intervalFactor":2,"legendFormat":"Error -{{instance}} ","refId":"A","step":60},{"expr":"sum by(instance) (rate(apiserver_request_count[5m]))","format":"time_series","intervalFactor":2,"legendFormat":"Request - {{instance}}","refId":"B","step":60}],"thresholds":[],"timeFrom":null,"timeShift":null,"title":"API Server Request Rates","tooltip":{"msResolution":false,"shared":true,"sort":0,"value_type":"individual"},"type":"graph","xaxis":{"buckets":null,"mode":"time","name":null,"show":true,"values":[]},"yaxes":[{"format":"short","logBase":1,"show":true},{"format":"short","logBase":1,"show":true}],"yaxis":{"align":false,"alignLevel":null}},{"aliasColors":{},"bars":false,"dashLength":10,"dashes":false,"datasource":"RANCHER_MONITORING","editable":true,"error":false,"fill":1,"grid":{"threshold1Color":"rgba(216, 200, 27, 0.27)","threshold2Color":"rgba(234, 112, 112, 0.22)"},"gridPos":{"h":7,"w":12,"x":12,"y":0},"id":7,"isNew":false,"legend":{"alignAsTable":false,"avg":false,"current":false,"hideEmpty":false,"hideZero":false,"max":false,"min":false,"rightSide":false,"show":true,"total":false,"values":false},"lines":true,"linewidth":1,"links":[],"nullPointMode":"null","percentage":false,"pointradius":5,"points":false,"renderer":"flot","seriesOverrides":[],"spaceLength":10,"stack":false,"steppedLine":false,"targets":[{"expr":"avg(apiserver_request_latencies_sum / apiserver_request_latencies_count) by (instance)","format":"time_series","intervalFactor":2,"legendFormat":"","refId":"A","step":30}],"thresholds":[],"timeFrom":null,"timeShift":null,"title":"API Server Request Latency","tooltip":{"msResolution":false,"shared":true,"sort":0,"value_type":"individual"},"type":"graph","xaxis":{"buckets":null,"mode":"time","name":null,"show":true,"values":[]},"yaxes":[{"format":"ns","logBase":1,"show":true},{"format":"short","logBase":1,"show":true}],"yaxis":{"align":false,"alignLevel":null}},{"aliasColors":{},"bars":false,"dashLength":10,"dashes":false,"datasource":"RANCHER_MONITORING","fill":1,"gridPos":{"h":9,"w":12,"x":0,"y":7},"id":11,"legend":{"avg":false,"current":false,"max":false,"min":false,"show":true,"total":false,"values":false},"lines":true,"linewidth":1,"links":[],"nullPointMode":"null","percentage":false,"pointradius":5,"points":false,"renderer":"flot","seriesOverrides":[],"spaceLength":10,"stack":false,"steppedLine":false,"targets":[{"expr":"increase(nginx_ingress_controller_nginx_process_requests_total[1m])","format":"time_series","intervalFactor":1,"legendFormat":"Request Rate","refId":"A"}],"thresholds":[],"timeFrom":null,"timeShift":null,"title":"Nginx Ingress Request Rate","tooltip":{"shared":true,"sort":0,"value_type":"individual"},"type":"graph","xaxis":{"buckets":null,"mode":"time","name":null,"show":true,"values":[]},"yaxes":[{"format":"short","label":null,"logBase":1,"max":null,"min":null,"show":true},{"format":"short","label":null,"logBase":1,"max":null,"min":null,"show":true}],"yaxis":{"align":false,"alignLevel":null}},{"aliasColors":{},"bars":false,"dashLength":10,"dashes":false,"datasource":"RANCHER_MONITORING","fill":1,"gridPos":{"h":9,"w":12,"x":12,"y":7},"id":9,"legend":{"avg":false,"current":false,"max":false,"min":false,"show":true,"total":false,"values":false},"lines":true,"linewidth":1,"links":[],"nullPointMode":"null","percentage":false,"pointradius":5,"points":false,"renderer":"flot","seriesOverrides":[],"spaceLength":10,"stack":false,"steppedLine":false,"targets":[{"expr":"sum(service_depth)","format":"time_series","intervalFactor":1,"legendFormat":"Service Depth","refId":"A"},{"expr":"sum(volumes_depth)","format":"time_series","intervalFactor":1,"legendFormat":"Volumes Depth","refId":"B"},{"expr":"sum(replicationmanager_depth)","format":"time_series","intervalFactor":1,"legendFormat":"Replication Manager Depth","refId":"C"},{"expr":"sum(statefulset_depth)","format":"time_series","intervalFactor":1,"legendFormat":"StatefulSet Depth","refId":"D"},{"expr":"sum(serviceaccount_depth)","format":"time_series","intervalFactor":1,"legendFormat":"Service Account Depth","refId":"E"},{"expr":"sum(endpoint_depth)","format":"time_series","intervalFactor":1,"legendFormat":"Endpoint Depth","refId":"F"},{"expr":"sum(deployment_depth)","format":"time_series","intervalFactor":1,"legendFormat":"Deployment Depth","refId":"G"},{"expr":"sum(daemonset_depth)","format":"time_series","intervalFactor":1,"legendFormat":"DaemonSet Depth","refId":"H"},{"expr":"sum(replicaset_depth)","format":"time_series","intervalFactor":1,"refId":"I"}],"thresholds":[],"timeFrom":null,"timeShift":null,"title":"Controller Manager Queue Depth","tooltip":{"shared":true,"sort":0,"value_type":"individual"},"type":"graph","xaxis":{"buckets":null,"mode":"time","name":null,"show":true,"values":[]},"yaxes":[{"format":"short","label":null,"logBase":1,"max":null,"min":null,"show":true},{"format":"short","label":null,"logBase":1,"max":null,"min":null,"show":true}],"yaxis":{"align":false,"alignLevel":null}}],"refresh":false,"schemaVersion":16,"style":"dark","tags":[],"templating":{"list":[]},"time":{"from":"now-5m","to":"now"},"timepicker":{"refresh_intervals":["5s","10s","30s","1m","5m","15m","30m","1h","2h","1d"],"time_options":["5m","15m","1h","6h","12h","24h","2d","7d","30d"]},"timezone":"browser","title":"Kubernetes Components","uid":"Ld4acTYmz","version":1}
\ No newline at end of file
+{"__requires":[{"type":"grafana","id":"grafana","name":"Grafana","version":"5.3.0"},{"type":"panel","id":"graph","name":"Graph","version":"5.0.0"},{"type":"datasource","id":"prometheus","name":"Prometheus","version":"5.0.0"}],"annotations":{"list":[{"builtIn":1,"datasource":"-- Grafana --","enable":true,"hide":true,"iconColor":"rgba(0, 211, 255, 1)","name":"Annotations & Alerts","type":"dashboard"}]},"description":"-","editable":true,"gnetId":5508,"graphTooltip":0,"id":null,"links":[],"panels":[{"aliasColors":{},"bars":false,"dashLength":10,"dashes":false,"datasource":"RANCHER_MONITORING","editable":true,"error":false,"fill":1,"grid":{"threshold1Color":"rgba(216, 200, 27, 0.27)","threshold2Color":"rgba(234, 112, 112, 0.22)"},"gridPos":{"h":7,"w":12,"x":0,"y":0},"id":6,"isNew":false,"legend":{"alignAsTable":false,"avg":false,"current":false,"hideEmpty":false,"hideZero":false,"max":false,"min":false,"rightSide":false,"show":true,"total":false,"values":false},"lines":true,"linewidth":1,"links":[],"nullPointMode":"null","percentage":false,"pointradius":5,"points":false,"renderer":"flot","seriesOverrides":[],"spaceLength":10,"stack":false,"steppedLine":false,"targets":[{"expr":"sum by(instance) (rate(apiserver_request_count{code!~\"2..\"}[5m]))","format":"time_series","intervalFactor":2,"legendFormat":"Error -{{instance}} ","refId":"A","step":60},{"expr":"sum by(instance) (rate(apiserver_request_count[5m]))","format":"time_series","intervalFactor":2,"legendFormat":"Request - {{instance}}","refId":"B","step":60}],"thresholds":[],"timeFrom":null,"timeShift":null,"title":"API Server Request Rates","tooltip":{"msResolution":false,"shared":true,"sort":0,"value_type":"individual"},"type":"graph","xaxis":{"buckets":null,"mode":"time","name":null,"show":true,"values":[]},"yaxes":[{"format":"short","logBase":1,"show":true},{"format":"short","logBase":1,"show":true}],"yaxis":{"align":false,"alignLevel":null}},{"aliasColors":{},"bars":false,"dashLength":10,"dashes":false,"datasource":"RANCHER_MONITORING","editable":true,"error":false,"fill":1,"grid":{"threshold1Color":"rgba(216, 200, 27, 0.27)","threshold2Color":"rgba(234, 112, 112, 0.22)"},"gridPos":{"h":7,"w":12,"x":12,"y":0},"id":7,"isNew":false,"legend":{"alignAsTable":false,"avg":false,"current":false,"hideEmpty":false,"hideZero":false,"max":false,"min":false,"rightSide":false,"show":true,"total":false,"values":false},"lines":true,"linewidth":1,"links":[],"nullPointMode":"null","percentage":false,"pointradius":5,"points":false,"renderer":"flot","seriesOverrides":[],"spaceLength":10,"stack":false,"steppedLine":false,"targets":[{"expr":"avg(apiserver_request_latencies_sum / apiserver_request_latencies_count) by (instance)","format":"time_series","intervalFactor":2,"legendFormat":"","refId":"A","step":30}],"thresholds":[],"timeFrom":null,"timeShift":null,"title":"API Server Request Latency","tooltip":{"msResolution":false,"shared":true,"sort":0,"value_type":"individual"},"type":"graph","xaxis":{"buckets":null,"mode":"time","name":null,"show":true,"values":[]},"yaxes":[{"format":"ns","logBase":1,"show":true},{"format":"short","logBase":1,"show":true}],"yaxis":{"align":false,"alignLevel":null}},{"aliasColors":{},"bars":false,"dashLength":10,"dashes":false,"datasource":"RANCHER_MONITORING","fill":1,"gridPos":{"h":9,"w":12,"x":0,"y":7},"id":11,"legend":{"avg":false,"current":false,"max":false,"min":false,"show":true,"total":false,"values":false},"lines":true,"linewidth":1,"links":[],"nullPointMode":"null","percentage":false,"pointradius":5,"points":false,"renderer":"flot","seriesOverrides":[],"spaceLength":10,"stack":false,"steppedLine":false,"targets":[{"expr":"increase(nginx_ingress_controller_nginx_process_requests_total[2m])","format":"time_series","intervalFactor":1,"legendFormat":"Request Rate","refId":"A"}],"thresholds":[],"timeFrom":null,"timeShift":null,"title":"Nginx Ingress Request Rate","tooltip":{"shared":true,"sort":0,"value_type":"individual"},"type":"graph","xaxis":{"buckets":null,"mode":"time","name":null,"show":true,"values":[]},"yaxes":[{"format":"short","label":null,"logBase":1,"max":null,"min":null,"show":true},{"format":"short","label":null,"logBase":1,"max":null,"min":null,"show":true}],"yaxis":{"align":false,"alignLevel":null}},{"aliasColors":{},"bars":false,"dashLength":10,"dashes":false,"datasource":"RANCHER_MONITORING","fill":1,"gridPos":{"h":9,"w":12,"x":12,"y":7},"id":9,"legend":{"avg":false,"current":false,"max":false,"min":false,"show":true,"total":false,"values":false},"lines":true,"linewidth":1,"links":[],"nullPointMode":"null","percentage":false,"pointradius":5,"points":false,"renderer":"flot","seriesOverrides":[],"spaceLength":10,"stack":false,"steppedLine":false,"targets":[{"expr":"sum(service_depth)","format":"time_series","intervalFactor":1,"legendFormat":"Service Depth","refId":"A"},{"expr":"sum(volumes_depth)","format":"time_series","intervalFactor":1,"legendFormat":"Volumes Depth","refId":"B"},{"expr":"sum(replicationmanager_depth)","format":"time_series","intervalFactor":1,"legendFormat":"Replication Manager Depth","refId":"C"},{"expr":"sum(statefulset_depth)","format":"time_series","intervalFactor":1,"legendFormat":"StatefulSet Depth","refId":"D"},{"expr":"sum(serviceaccount_depth)","format":"time_series","intervalFactor":1,"legendFormat":"Service Account Depth","refId":"E"},{"expr":"sum(endpoint_depth)","format":"time_series","intervalFactor":1,"legendFormat":"Endpoint Depth","refId":"F"},{"expr":"sum(deployment_depth)","format":"time_series","intervalFactor":1,"legendFormat":"Deployment Depth","refId":"G"},{"expr":"sum(daemonset_depth)","format":"time_series","intervalFactor":1,"legendFormat":"DaemonSet Depth","refId":"H"},{"expr":"sum(replicaset_depth)","format":"time_series","intervalFactor":1,"refId":"I"}],"thresholds":[],"timeFrom":null,"timeShift":null,"title":"Controller Manager Queue Depth","tooltip":{"shared":true,"sort":0,"value_type":"individual"},"type":"graph","xaxis":{"buckets":null,"mode":"time","name":null,"show":true,"values":[]},"yaxes":[{"format":"short","label":null,"logBase":1,"max":null,"min":null,"show":true},{"format":"short","label":null,"logBase":1,"max":null,"min":null,"show":true}],"yaxis":{"align":false,"alignLevel":null}}],"refresh":false,"schemaVersion":16,"style":"dark","tags":[],"templating":{"list":[]},"time":{"from":"now-5m","to":"now"},"timepicker":{"refresh_intervals":["5s","10s","30s","1m","5m","15m","30m","1h","2h","1d"],"time_options":["5m","15m","1h","6h","12h","24h","2d","7d","30d"]},"timezone":"browser","title":"Kubernetes Components","uid":"Ld4acTYmz","version":1}
\ No newline at end of file
diff -uNr charts/rancher-monitoring/v0.1.3/charts/prometheus/templates/prometheus.yaml charts/rancher-monitoring/v0.1.4/charts/prometheus/templates/prometheus.yaml
--- charts/rancher-monitoring/v0.1.3/charts/prometheus/templates/prometheus.yaml	2020-08-06 14:20:41.000000000 -0700
+++ charts/rancher-monitoring/v0.1.4/charts/prometheus/templates/prometheus.yaml	2020-09-17 17:58:14.000000000 -0700
@@ -84,7 +84,7 @@
         name: alertmanager-operated
         port: http
 {{- end }}
-  baseImage: {{ template "system_default_registry" . }}{{ .Values.image.repository }}
+  image: {{ template "system_default_registry" . }}{{ .Values.image.repository }}
 {{- if or (.Values.externalLabels) (eq .Values.level "project") }}
   externalLabels:
 {{- if and (eq .Values.level "project") }}
@@ -96,7 +96,7 @@
 {{- end }}
 {{- if .Values.thanos.enabled }}
   thanos: 
-    baseImage: {{ template "system_default_registry" . }}{{ .Values.image.thanos.repository }}
+    image: {{ template "system_default_registry" . }}{{ .Values.image.thanos.repository }}
     version: {{ .Values.image.thanos.tag }}
     objectStorageConfig:
       key: thanos.yaml
diff -uNr charts/rancher-monitoring/v0.1.3/questions.yml charts/rancher-monitoring/v0.1.4/questions.yml
--- charts/rancher-monitoring/v0.1.3/questions.yml	2020-08-06 14:20:41.000000000 -0700
+++ charts/rancher-monitoring/v0.1.4/questions.yml	2020-09-17 17:58:14.000000000 -0700
@@ -1 +1 @@
-rancher_min_version: 2.4.6-rc1
+rancher_min_version: v2.5.0-alpha1
```

**Comment by deniseschannon** [comment]
v0.1.3 is new to 2.5,so we don't need to bump to v0.1.4 yet as we only want 1 version of monitoring to be released with v2.5

**Comment by aiyengar2** [comment]
> v0.1.3 is new to 2.5,so we don't need to bump to v0.1.4 yet as we only want 1 version of monitoring to be released with v2.5

PR has been modified to make changes in place to v0.1.3 instead of creating a new chart.

File: charts/rancher-monitoring/v0.1.3/values.yaml
Comment (aiyengar2) [inline charts/rancher-monitoring/v0.1.3/values.yaml]: Can be dropped now since https://github.com/rancher/system-charts/pull/341 already addresses this.

**Review by pennyscissors — APPROVED** [review]
lgtm

File: charts/rancher-monitoring/v0.1.3/questions.yml
Comment (aiyengar2) [inline charts/rancher-monitoring/v0.1.3/questions.yml:1]: @deniseschannon and I chatted offline about this before, we want to bump this `rancher_min_version` until QA validates it onto 2.4; once that is done, this min version will be updated on merge to `release-v2.4`.

cc: @PennyScissors

**Review by dramich — APPROVED** [review]

**Pull request merged**
